packages/grafana-data/src/types/datasource.ts

```diff
@@ -549,7 +549,7 @@ export interface DataSourceSettings<T extends DataSourceJsonData = DataSourceJso
   name: string;
   typeLogoUrl: string;
   type: string;
-  typeName: string;
+  typeName?: string;
```

Comment:
it's not optional

Reply:
Hmm, there are a lot of type errors in other components due to the missing `typeName`.